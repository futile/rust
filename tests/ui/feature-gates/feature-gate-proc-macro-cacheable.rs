//@ force-host
//@ no-prefer-dynamic

// #![feature(proc_macro_cacheable)]

#![crate_type = "proc-macro"]

extern crate proc_macro;

use proc_macro::TokenStream;

#[proc_macro_derive(SomeMacro)]
#[proc_macro_cacheable] //~ error: the `#[proc_macro_cacheable]` attribute is an experimental feature
pub fn derive(input: TokenStream) -> TokenStream {
    "".parse().unwrap()
}
