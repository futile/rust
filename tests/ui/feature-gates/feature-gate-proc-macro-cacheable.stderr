error[E0658]: the `#[proc_macro_cacheable]` attribute is an experimental feature
  --> $DIR/feature-gate-proc-macro-cacheable.rs:13:1
   |
LL | #[proc_macro_cacheable]
   | ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add `#![feature(proc_macro_cacheable)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0658`.
